java/src/com/ibm/streamsx/topology/spl/SPLValue.java

```diff
@@ -0,0 +1,151 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2015  
+ */
+package com.ibm.streamsx.topology.spl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.ibm.json.java.JSONObject;
+import com.ibm.json.java.OrderedJSONObject;
+import com.ibm.streams.operator.Type;
+import com.ibm.streams.operator.Type.MetaType;
+import com.ibm.streamsx.topology.generator.spl.SPLGenerator;
+
+/**
+ * An implementation private wrapper for values of any SPL type. 
+ * <p>
+ * See {@link SPL#createParamValue(Object, MetaType)
+ *
+ * @param <T> the SPLValue's type
+ */
+class SPLValue<T> {
+    private static Map<MetaType,Info> mtypeMap = new HashMap<>();
+    static {
+        mtypeMap.put(MetaType.RSTRING, new Info(String.class, ""));
+        mtypeMap.put(MetaType.USTRING, new Info(String.class, "", "utf16"));
+        mtypeMap.put(MetaType.BOOLEAN, new Info(Boolean.class, false));
+        mtypeMap.put(MetaType.INT8,    new Info(Byte.class, (byte)0));
+        mtypeMap.put(MetaType.UINT8,   new Info(Byte.class, (byte)0, "unsigned"));
+        mtypeMap.put(MetaType.INT16,   new Info(Short.class, (short)0));
+        mtypeMap.put(MetaType.UINT16,  new Info(Short.class, (short)0, "unsigned"));
+        mtypeMap.put(MetaType.INT32,   new Info(Integer.class, 0));
+        mtypeMap.put(MetaType.UINT32,  new Info(Integer.class, 0, "unsigned"));
+        mtypeMap.put(MetaType.INT64,   new Info(Long.class, (long)0));
+        mtypeMap.put(MetaType.UINT64,  new Info(Long.class, (long)0, "unsigned"));
+        mtypeMap.put(MetaType.FLOAT32, new Info(Float.class, (float)0.0));
+        mtypeMap.put(MetaType.FLOAT64, new Info(Double.class, (double)0.0));
+        // add more as needed
+    }
+    
+    private static class Info {
```

Comment:
Not sure of the value of this, can we not use the type system provided by the operator API? It has the ability to get the default value for an SPL type.